crates/subspace-farmer/src/bin/subspace-farmer/bench_rpc_client.rs

```diff
@@ -90,24 +98,25 @@ impl RpcClient for BenchRpcClient {
 
     async fn subscribe_archived_segments(
         &self,
-    ) -> Result<mpsc::Receiver<ArchivedSegment>, MockError> {
-        let (sender, receiver) = mpsc::channel(10);
+    ) -> Result<Pin<Box<dyn Stream<Item = ArchivedSegment> + Send + 'static>>, MockError> {
+        let (mut sender, receiver) = mpsc::channel(10);
         let archived_segments_receiver = self.inner.archived_segments_receiver.clone();
         tokio::spawn(async move {
-            while let Some(archived_segment) = archived_segments_receiver.lock().await.recv().await
+            while let Some(archived_segment) = archived_segments_receiver.lock().await.next().await
             {
                 if sender.send(archived_segment).await.is_err() {
                     break;
                 }
             }
         });
 
-        Ok(receiver)
+        Ok(Box::pin(receiver))
     }
 
     async fn acknowledge_archived_segment(&self, segment_index: u64) -> Result<(), MockError> {
         self.inner
             .acknowledge_archived_segment_sender
+            .clone()
```

Comment:
This looks kinda wrong. Maybe we can use smth like [`tokio-stream`](https://docs.rs/tokio-stream/latest/tokio_stream/) and preserve tokio mpsc channels?

Reply:
It needs `&mut` to actually call `send`, but `acknowledge_archived_segment` only has `&self`.

Reply:
Do you think it is worth it comparing to `.clone()`? It is really weird that tokio's channels don't implement `Stream`.

Reply:
I guess they won't implement it till it lands to the standart library. That's why they have separate crate with wrappers which implement `futures` traits (they are zero cost).

I guess my concern is just consistency. If we use mpsc channels it is weird to use different ones from different crates 🤔

Reply:
Well, consistency is to not use Tokio channels :upside_down_face: 

I don't have too strong of an opinion here, but channels as is work fine. The cost of clone appears to be just 2 atomic operations, so unlikely to be a bottleneck.

Reply:
Yeah, true